src/pcre2_compile.c

```diff
@@ -11022,17 +11068,17 @@ there are no occurrences of *PRUNE or *SKIP (though there is an option to
 disable this case). */
 
 if ((re->overall_options & PCRE2_ANCHORED) == 0 &&
-     is_anchored(codestart, 0, &cb, 0, FALSE))
+     is_anchored(codestart, 0, &cb, 0, FALSE, (optim_flags & PCRE2_OPTIM_DOTSTAR_ANCHOR) != 0))
```

Comment:
by adding the != 0 it become a little longer so maybe worth getting it in the next line since the other options were already indented anyway?

Reply:
OK. I split the long expression out to a local variable assignment rather than making the `if` condition span 3 lines.